src/providers/pkcs11/key_management.rs

```diff
@@ -228,38 +266,41 @@ impl Provider {
             return Err(ResponseStatus::PsaErrorInvalidArgument);
         }
 
-        template.push(Attribute::Class(ObjectClass::PUBLIC_KEY));
-        template.push(Attribute::KeyType(KeyType::RSA));
-        template.push(Attribute::Token(true.into()));
         template.push(Attribute::Modulus(modulus_object.into()));
         template.push(Attribute::PublicExponent(exponent_object.into()));
-        template.push(Attribute::Verify(true.into()));
         template.push(Attribute::Encrypt(true.into()));
-        template.push(Attribute::Id(key_id.to_be_bytes().to_vec()));
         template.push(Attribute::Private(false.into()));
         template.push(Attribute::AllowedMechanisms(vec![MechanismType::RSA_PKCS]));
+        template.push(Attribute::KeyType(KeyType::RSA));
 
-        trace!("CreateObject command");
-        match session.create_object(&template) {
-            Ok(key) => {
-                if let Err(e) =
-                    self.key_info_store
-                        .insert_key_info(key_triple, &key_id, key_attributes)
-                {
-                    format_error!("Failed to insert the mappings, deleting the key.", e);
-                    if let Err(e) = session.destroy_object(key) {
-                        format_error!("Failed to destroy public key: ", e);
-                    }
-                    Err(e)
-                } else {
-                    Ok(psa_import_key::Result {})
-                }
-            }
-            Err(error) => {
-                format_error!("Import key status: ", error);
-                Err(to_response_status(error))
-            }
+        Ok(())
+    }
+
+    pub(super) fn handle_ecc_public_import_attrib(
+        &self,
+        key_data: &[u8],
+        bits: usize,
+        curve_family: EccFamily,
+        template: &mut Vec<Attribute>,
+    ) -> Result<()> {
+        let mut data_in = key_data.to_vec();
+        if data_in.len() > 255 {
+            return Err(ResponseStatus::PsaErrorInvalidArgument);
         }
+        let mut ec_point = vec![0x04, data_in.len() as u8];
+        ec_point.append(&mut data_in);
```

Comment:
If `data_in` is copied here, it might be worth to zeroize after

Reply:
Not sure it's really needed, since this can only be a public key that we're handling here, but I'll zeroize at the end, just to be sure.

Reply:
Hmm, actually I'd have to start searching in the `template` vector after we're done creating the key for that `EcPoint`, since this `ec_point` is essentially moved into that one. There's similar processing with no zeroizing for the RSA keys, so I'll just let this be, it's not sensitive stuff.